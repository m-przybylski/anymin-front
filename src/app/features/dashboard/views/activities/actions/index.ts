import * as ActivitiesActions from './activities.actions';
import * as ActivitiesApiActions from './activities-api.actions';
import * as ActivitiesPageActions from './activities-page.actions';
import * as BalanceApiActions from './balance-api.actions';
import * as ActivitiesWsActions from './activities-ws.actions';

export { ActivitiesActions, ActivitiesApiActions, ActivitiesPageActions, BalanceApiActions, ActivitiesWsActions };
