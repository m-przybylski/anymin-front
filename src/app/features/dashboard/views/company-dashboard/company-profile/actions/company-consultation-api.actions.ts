// tslint:disable:max-classes-per-file
import { Action } from '@ngrx/store';
import { ICompanyConsultationDetails, ICompanyEmployeeRowComponent } from '../services/company-consultation.service';

export enum CompanyConsultationApiActionTypes {
  LoadSuccess = '[Comp consult API] Load consultation success',
  LoadFailure = '[Comp consult API] Load consultation failure',
  LoadInvitesSuccess = '[Comp consult API] Load invites success',
  LoadInvitesFailure = '[Comp consult API] Load invites failure',
  DeleteEmploymentSuccess = '[Comp consult API] Delete employment success',
  DeleteEmploymentFailure = '[Comp consult API] Delete employment failure',
  DeleteInviteSuccess = '[Comp consult API] Delete invite success',
  DeleteInviteFailure = '[Comp consult API] Delete invite failure',
}

export class LoadConsultationSuccessAction implements Action {
  public readonly type = CompanyConsultationApiActionTypes.LoadSuccess;

  constructor(public payload: ICompanyConsultationDetails) {}
}

export class LoadConsultationFailureAction implements Action {
  public readonly type = CompanyConsultationApiActionTypes.LoadFailure;

  constructor(public payload: any) {}
}

export class LoadInvitesSuccessAction implements Action {
  public readonly type = CompanyConsultationApiActionTypes.LoadInvitesSuccess;

  constructor(public payload: ReadonlyArray<ICompanyEmployeeRowComponent>) {}
}

export class LoadInvitesFailureAction implements Action {
  public readonly type = CompanyConsultationApiActionTypes.LoadInvitesFailure;

  constructor(public payload: any) {}
}

export class DeleteEmploymentSuccessAction implements Action {
  public readonly type = CompanyConsultationApiActionTypes.DeleteEmploymentSuccess;

  constructor(public payload: string) {}
}

export class DeleteEmploymentFailureAction implements Action {
  public readonly type = CompanyConsultationApiActionTypes.DeleteEmploymentFailure;

  constructor(public payload: any) {}
}

export class DeleteInviteSuccessAction implements Action {
  public readonly type = CompanyConsultationApiActionTypes.DeleteInviteSuccess;

  constructor(public payload: string) {}
}

export class DeleteInviteFailureAction implements Action {
  public readonly type = CompanyConsultationApiActionTypes.DeleteInviteFailure;

  constructor(public payload: any) {}
}

export type CompanyConsultationApiActionsUnion =
  | LoadConsultationSuccessAction
  | LoadConsultationFailureAction
  | LoadInvitesSuccessAction
  | LoadInvitesFailureAction
  | DeleteEmploymentSuccessAction
  | DeleteEmploymentFailureAction;
