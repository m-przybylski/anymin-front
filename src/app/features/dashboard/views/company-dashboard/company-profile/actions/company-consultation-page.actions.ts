// tslint:disable:max-classes-per-file
import { Action } from '@ngrx/store';
import { GetSessionWithAccount } from '@anymind-ng/api';
import { UserTypeEnum } from '@platform/core/reducers/navbar.reducer';

export enum CompanyConsultationPageActionTypes {
  Load = '[Comp consult page] Load consultation',
  LoadInvites = '[Comp consult page] Load pending invites',
  Clear = '[Comp consult page] Clear',
  DeleteEmployment = '[Comp consult page] Delete employment',
  DeleteInvite = '[Comp consult page] Delete invite',
}

export class LoadConsultationsAction implements Action {
  public readonly type = CompanyConsultationPageActionTypes.Load;

  constructor(
    public payload: {
      serviceId: string;
      getSessionWithAccount?: GetSessionWithAccount;
      userType?: UserTypeEnum;
    },
  ) {}
}

export class LoadPendingInvitesAction implements Action {
  public readonly type = CompanyConsultationPageActionTypes.LoadInvites;

  constructor(public payload: string) {}
}

export class ClearAction implements Action {
  public readonly type = CompanyConsultationPageActionTypes.Clear;
}

export class DeleteEmploymentAction implements Action {
  public readonly type = CompanyConsultationPageActionTypes.DeleteEmployment;

  constructor(public payload: string) {}
}

export class DeleteInviteAction implements Action {
  public readonly type = CompanyConsultationPageActionTypes.DeleteInvite;

  constructor(public payload: string) {}
}

export type CompanyConsultationPageActionsUnion =
  | LoadConsultationsAction
  | LoadPendingInvitesAction
  | ClearAction
  | DeleteEmploymentAction
  | DeleteInviteAction;
