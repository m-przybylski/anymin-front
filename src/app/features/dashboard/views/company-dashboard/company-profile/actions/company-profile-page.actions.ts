// tslint:disable:max-classes-per-file
import { Action } from '@ngrx/store';

export enum CompanyProfilePageActionTypes {
  LeaveConsultation = '[Comp profile page] Leave consultation',
  RemoveExpertFromConsultation = '[Comp profile page] Remove expert from consultation',
}

export class LeaveConsultationAction implements Action {
  public readonly type = CompanyProfilePageActionTypes.LeaveConsultation;
  constructor(public payload: string) {}
}

export class RemoveExpertFromConsultationAction implements Action {
  public readonly type = CompanyProfilePageActionTypes.RemoveExpertFromConsultation;
  constructor(public payload: string) {}
}

export type CompanyProfilePageActionsUnion = LeaveConsultationAction | RemoveExpertFromConsultationAction;
