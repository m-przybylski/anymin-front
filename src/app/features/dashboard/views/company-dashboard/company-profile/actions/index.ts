import * as CompanyProfileApiActions from './company-profile-api.actions';
import * as CompanyProfilePageActions from './company-profile-page.actions';

export { CompanyProfileApiActions, CompanyProfilePageActions };
