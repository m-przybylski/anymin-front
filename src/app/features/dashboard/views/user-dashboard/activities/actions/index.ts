import * as ExpertActivitiesActions from './expert-activities.actions';
import * as ExpertActivitiesApiActions from './expert-activities-api.actions';
import * as ExpertActivitiesPageActions from './expert-activities-page.actions';
import * as BalanceApiActions from './balance-api.actions';
import * as ExpertActivitiesWsActions from './expert-activities-ws.actions';

export {
  ExpertActivitiesActions,
  ExpertActivitiesApiActions,
  ExpertActivitiesPageActions,
  BalanceApiActions,
  ExpertActivitiesWsActions,
};
