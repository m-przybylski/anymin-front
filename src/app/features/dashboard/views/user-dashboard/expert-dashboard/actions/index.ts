import * as ExpertDashboardActions from './expert-dashboard.actions';
import * as ExpertDashboardApiActions from './expert-dashboard.api.actions';

export { ExpertDashboardActions, ExpertDashboardApiActions };
