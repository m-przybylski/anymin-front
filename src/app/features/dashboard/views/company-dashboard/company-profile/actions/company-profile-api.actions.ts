// tslint:disable:max-classes-per-file
import { Action } from '@ngrx/store';
import { IOrganizationProfile } from '../services/company-profile.service';
import { IExpertCompanyDashboardResolverData } from '../../../common/resolver-helpers';

export enum CompanyProfileApiActionTypes {
  LoadProfileActionSuccess = '[Comp profile API] Load profile success',
  LoadProfileActionFailure = '[Comp profile API] Load profile failure',
  DeleteEmploymentSuccess = '[Comp profile API] Delete employment success',
  DeleteEmploymentFailure = '[Comp profile API] Delete employment failure',
}

export class LoadProfileActionSuccess implements Action {
  public readonly type = CompanyProfileApiActionTypes.LoadProfileActionSuccess;

  constructor(public payload: IExpertCompanyDashboardResolverData<IOrganizationProfile>) {}
}

export class LoadProfileActionFailure implements Action {
  public readonly type = CompanyProfileApiActionTypes.LoadProfileActionFailure;

  constructor(public payload: IOrganizationProfile) {}
}

export class DeleteEmploymentSuccessAction implements Action {
  public readonly type = CompanyProfileApiActionTypes.DeleteEmploymentSuccess;
  constructor(public payload: string) {}
}

export class DeleteEmploymentFailureAction implements Action {
  public readonly type = CompanyProfileApiActionTypes.DeleteEmploymentFailure;
  constructor(public payload: any) {}
}

export type CompanyProfileApiActionsUnion =
  | LoadProfileActionSuccess
  | LoadProfileActionFailure
  | DeleteEmploymentSuccessAction
  | DeleteEmploymentFailureAction;
