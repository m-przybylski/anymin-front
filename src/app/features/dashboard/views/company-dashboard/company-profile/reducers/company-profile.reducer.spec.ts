import { reducer, IState } from './index';
import { CompanyProfileApiActions, CompanyProfilePageActions } from '../actions';

const initialState: IState = {
  isLoading: false,
};

describe('company profile reducer', () => {
  describe('undefined action', () => {
    it('should return the default state', () => {
      const result = reducer(undefined, {} as any);

      expect(result).toMatchSnapshot();
    });
  });

  describe('DeleteEmploymentSuccessAction action', () => {
    it('should remove expert from consultation', () => {
      const state = {
        isLoading: false,
        organizationProfile: {
          profile: {
            organization: {
              services: [
                {
                  service: {},
                  employments: [{ id: '1' }, { id: '2' }, { id: '3' }],
                },
                {
                  service: {},
                  employments: [{ id: '5' }, { id: '6' }, { id: '7' }],
                },
              ],
            },
          },
        },
      } as any;
      const result = reducer(state, new CompanyProfileApiActions.DeleteEmploymentSuccessAction('7'));
      expect(result).toMatchSnapshot();
    });
    it('should not throw when no data to remove', () => {
      const state = { isLoading: false } as any;
      const result = reducer(state, new CompanyProfileApiActions.DeleteEmploymentSuccessAction('7'));
      expect(result).toMatchSnapshot();
    });
  });
  describe('LoadProfileAction', () => {
    it('should start loading profile', () => {
      const result = reducer(initialState, new CompanyProfilePageActions.LoadProfileAction('fake profile'));
      expect(result).toMatchSnapshot();
    });
  });
  describe('LoadProfileActionSuccess', () => {
    it('should start loading profile', () => {
      const profile: any = '👍👍👍';
      const result = reducer({ isLoading: true }, new CompanyProfileApiActions.LoadProfileActionSuccess(profile));
      expect(result).toMatchSnapshot();
    });
  });
  describe('LoadProfileActionFailure', () => {
    it('should start loading profile', () => {
      const profile: any = '👍👍👍';
      const result = reducer({ isLoading: true }, new CompanyProfileApiActions.LoadProfileActionFailure(profile));
      expect(result).toMatchSnapshot();
    });
  });

  describe('UpdateProfileAction', () => {
    it('should start loading profile', () => {
      const result = reducer(initialState, new CompanyProfilePageActions.UpdateProfileAction('fake profile'));
      expect(result).toMatchSnapshot();
    });
  });
  describe('AddConsultation', () => {
    it('should append new consultation', () => {
      const state = {
        isLoading: false,
        organizationProfile: {
          profile: {
            organization: {
              services: ['fake service 1'],
            },
          },
        },
      } as any;
      const newConsultation: any = 'fake service 2';
      const result = reducer(state, new CompanyProfilePageActions.AddConsultation(newConsultation));
      expect(result.organizationProfile && result.organizationProfile.profile.organization.services).toHaveLength(2);
    });
  });
});
