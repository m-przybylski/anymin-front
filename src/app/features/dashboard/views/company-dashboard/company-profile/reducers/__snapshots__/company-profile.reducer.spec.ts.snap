// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`company profile reducer DeleteEmploymentSuccessAction action should not throw when no data to remove 1`] = `
Object {
  "isLoading": false,
  "organizationProfile": undefined,
}
`;

exports[`company profile reducer DeleteEmploymentSuccessAction action should remove expert from consultation 1`] = `
Object {
  "isLoading": false,
  "organizationProfile": Object {
    "profile": Object {
      "organization": Object {
        "services": Array [
          Object {
            "employments": Array [
              Object {
                "id": "1",
              },
              Object {
                "id": "2",
              },
              Object {
                "id": "3",
              },
            ],
            "service": Object {},
          },
          Object {
            "employments": Array [
              Object {
                "id": "5",
              },
              Object {
                "id": "6",
              },
            ],
            "service": Object {},
          },
        ],
      },
    },
  },
}
`;

exports[`company profile reducer LoadProfileAction should start loading profile 1`] = `
Object {
  "error": undefined,
  "isLoading": true,
  "organizationProfile": undefined,
}
`;

exports[`company profile reducer LoadProfileActionFailure should start loading profile 1`] = `
Object {
  "error": "üëçüëçüëç",
  "isLoading": false,
}
`;

exports[`company profile reducer LoadProfileActionSuccess should start loading profile 1`] = `
Object {
  "error": undefined,
  "isLoading": false,
  "organizationProfile": "üëçüëçüëç",
}
`;

exports[`company profile reducer UpdateProfileAction should start loading profile 1`] = `
Object {
  "error": undefined,
  "isLoading": true,
  "organizationProfile": undefined,
}
`;

exports[`company profile reducer undefined action should return the default state 1`] = `
Object {
  "isLoading": false,
}
`;
